### PR TITLE
Aankirz: Fixes interop-invalidyear

### PR DESCRIPTION
**Pull request by Aankirz**
## Description
Fixes [issue 2975](https://github.com/web-platform-tests/wpt.fyi/issues/2975)
- To improve the user experience
- Whenever user enters any invalid year (Interop) , redirects to a sorry page.

## Review Information
- Try entering any invalid year to https://wpt.fyi/interop-XXXX

## Changes
- Added a new file `interop-404.js` in `webapp/components`
- Modified `interop_handler.go`

### CODE REVIEW

File: webapp/interop_handler.go [hunk 1]
```diff
@@ -60,5 +62,8 @@ func interopHandler(w http.ResponseWriter, r *http.Request) {
 		Embedded: embedded != nil && *embedded,
 		Year:     year,
 	}
+
+    
+
```
Comment (DanielRyanSmith) [inline webapp/interop_handler.go:67]: These new blank lines can be removed, or you can keep one if you'd like 🙂

File: webapp/components/interop-404.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        height: 60px;
+      }
+      h1 {
+        margin: 50px 0 50px 0;
+        font-weight: 300;
+      }
+    </style>
+    <div>
+      <iron-icon icon="error"></iron-icon>
+      <h1>Sorry, data is not available for this year 🥲</h1>
```
Comment (DanielRyanSmith) [inline webapp/components/interop-404.js:26]:
```suggestion
      <h1>Interop data is not available for this year</h1>
```

**Review by DanielRyanSmith — COMMENTED** [review]
Thanks for the fix @Aankirz! Can you provide a screenshot of the build working locally?

**Comment by Aankirz** [comment]
> Thanks for the fix @Aankirz! Can you provide a screenshot of the build working locally?

So sorry 😔, but I am getting installation issues so I wrote the code through logic (checked it line by line) and can't produce a local run, actually I asked it in my prev issue assigned, but I guess mentors might be busy.



Can you pls guide me, I followed every step ran script using DOCKER, but I'm getting these installation errors:  (After the step of "npm test" in webapp) My chromedriver path : "/opt/homebrew/bin/chromedriver", (Specs: Macos M1) I tried using chat gpt and googling, but was not able to resolve 😢 

Thanks for the Help ✨

Screenshot 2023-03-26 at 1 50 18 AM
4901-b8e5-fbad73c28468.png">

**Comment by Aankirz** [comment]
@DanielRyanSmith , can I pls get a link of the community where I can ask my doubts 🫥, Thanks a lott ✨

**Comment by Aankirz** [comment]
Should I open a issue regarding this ? 🙂

**Comment by Aankirz** [comment]
> Hey @Aankirz, this is a fine place to talk to us about questions you might have.
> 
> > Specs: Macos M1
> 
> Hmm, we have a known issue with running this project on M1 chips. I'll let you know if we come up with a solution for this.
Screenshot 2023-03-28 at 11 45 51 AM


Now it started with no errors, but when I go to localhost:8080, then the browser says 

'localhost didn’t send any data.
ERR_EMPTY_RESPONSE'

Can you pls help in this ? 🥲

**Comment by Aankirz** [comment]
@DanielRyanSmith Just to confirm is `npm test` the final command after which wpt.fyi website will be started on my localhost ??

**Comment by DanielRyanSmith** [comment]
@Aankirz unfortunately, I think you will have probably running the application locally until we investigate the ability to run for arm-based chips. [This issue was filed](https://github.com/web-platform-tests/wpt.fyi/issues/3196), so you are not the only person to run into this issue. I'm sorry, but I do not have an answer for how to fix this just yet.

**Comment by Aankirz** [comment]
@DanielRyanSmith , I changed my laptop and now it's a windows 11

and now getting this error (I'm using docker dev environment):
WhatsApp Image 2023-03-29 at 19 40 53

**Comment by Aankirz** [comment]
And just want to know what spec are you using to run this repo pls ? 🫥

**Comment by racheljay** [comment]
Hi @Aankirz! Can you tell me what you get when you post the following commands? Run `uname -m` and it will tell you the architecture of the computer you're currently using. If it says `arm64` that means you won't be able to run the environment on your computer because inotify is incompatible, and the environment requires this to run. If it's not `arm64` try running `apt search inotify-tools` and post the response you get.

**Comment by Aankirz** [comment]
No it's not arm64, I've changed my laptop and I tried the command:

Screenshot (3)


Thanks for your time @racheljay , but can you help me with this , This is the error I'm getting in my windows 11: inotifywait



Screenshot (4)

I also tried to install intoify tools using : `sudo apt-get install inotify-tools
`

But still getting this:
Screenshot (5)


Can you pls guide me  how did you installed, Thanks in advance

**Comment by Aankirz** [comment]
Thaaaaaaaaaaanks a lotttttt @racheljay 🥹,

 It's now working fine

**Pull request closed without merging**